### PR TITLE
Requeststatus

### PR DESCRIPTION
**Pull request by Fareedh-12**
## Naming Rules

Name your PR like this: ISSUENUMBER-LAPTOP-REQUEST-STATUS-FAREEDH

## Description

Write a summary of your change. Explain why you have made it.
I have created laptop-request-status view
I have create api endpoint GET api/laptop-donations/:id
I have created different request status to be displaying different things on the react page

## Related to
following on request status 

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1
created fake requests with different states
created end point for /laptop-request-status/<id> 
created different request states basing on the current given request state
created react component to view the status of the request

changed the title of the application from starter kit to laptop loop

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -1,14 +1,66 @@
...
 import { Router } from "express";
 import db from "./db";
 
 import logger from "./utils/logger";
 
 const router = Router();
 
+let requests = [
+	{
+		id: 1,
```
Comment (gregdyke) [inline server/api.js]: can you make the IDs like those I mentioned in the trello card? Instead of 1-5, we have 100, 110, 111, 120, 121, 130, 140

**Review by gregdyke — APPROVED** [review]
Very nice! I think this is actually ready to go! You can fix the ids as a followup when you add the address information from the laptop_donation

File: server/api.js (hunk 1 above)
Comment (Fareedh-12) [reply server/api.js]: @gregdyke please review my code. I have not understood why we need the task **GET /api/laptop_donation/<id> (can return full data as is basic query from db)**.
I have used fake data for the laptop donation, please checkout my implementation and give me some more advice.

File: client/src/pages/ViewRequestStatus/ButtonComponent.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+const ButtonComponent = (props) => {
```
Comment (gregdyke) [inline client/src/pages/ViewRequestStatus/ButtonComponent.js:3]: I'm not sure about this buttoncomponent - might be easier to just do a button - but let's find out :)

File: client/src/pages/ViewRequestStatus/RequestStatusView.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+					<ButtonComponent command="No thank you!" />
+				</>
+			);
+		} else if (request.status === "ACCEPTED") {
+			return (
+				<>
+					<p>
+						Please let us know when you have received your laptop / when you
+						have picked up your laptop at{" "}
+						<strong>
```
Comment (gregdyke) [inline client/src/pages/ViewRequestStatus/RequestStatusView.js:60]: good that we keep the address here

**Review by gregdyke — APPROVED** [review]

File: server/api.js (hunk 1 above)
Comment (Fareedh-12) [reply server/api.js]: @gregdyke I have added the api endpoint: Get /api/laptop-donations/:id
Please review my code and leave me some feedback.

File: server/api.js [hunk 4]
```diff
@@ -1,14 +1,118 @@
...
+		phoneNumber: "073820384924",
+		status: "FULLFILLED",
+	},
+];
+
+router.get("/laptop-donation/:id", async (req, res) => {
+	 console.log("here");
+		try {
+			const result = await db.query("SELECT * from laptop_donation WHERE id = $1", [req.params.id]);
+			res.json(result.rows[0]);
```
Comment (gregdyke) [inline server/api.js]: the mapping to camelCase names needs to happen on result.rows

**Review by gregdyke — APPROVED** [review]

**Comment by gregdyke** [comment]
Hi Fareed. I would merge this change, so that followups can be done. First would be to have the GET laptop_donation/:id return camel case JSON + remove the full inclusion of the laptopDonation/laptopAssignment object from the fake data (using the GET laptop_donation/:id to get the address if needed in a .then after fetching the request)

File: server/api.js [hunk 5]
```diff
@@ -1,14 +1,118 @@
...
+		id: 140,
+		firstName: "Don",
+		lastName: "Yen",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "FULLFILLED",
+	},
+];
+
+router.get("/laptop-donation/:id", async (req, res) => {
```
Comment (gregdyke) [inline server/api.js]: that's right, this should be laptop_donation

File: server/api.js [hunk 6]
```diff
@@ -1,14 +1,118 @@
...
+	 console.log("here");
+		try {
+			const result = await db.query("SELECT * from laptop_donation WHERE id = $1", [req.params.id]);
+			res.json(result.rows[0]);
+		} catch (e) {
+			console.error(e);
+			res.sendStatus(400);
+		}
+});
+router.get("/laptop-request-status/:id", (req, res) => {
```
Comment (gregdyke) [inline server/api.js]: this should be laptop_request/:id - it's part of the laptop_request collection of endpoints

**Review by gregdyke — CHANGES_REQUESTED** [review]

File: server/api.js (hunk 6 above)
Comment (Fareedh-12) [reply server/api.js]: corrected!

File: server/api.js (hunk 5 above)
Comment (Fareedh-12) [reply server/api.js]: corrected!

File: server/api.js (hunk 4 above)
Comment (Fareedh-12) [reply server/api.js]: corrected

**Review by gregdyke — APPROVED** [review]

**Pull request merged**